src/test/scala/karazin/scala/users/group/week2/homework/HomeworkSpecification.scala

```diff
@@ -50,28 +50,29 @@ object HomeworkSpecification extends Properties("Homework"):
   }
 
   property("negation") = forAll { (rational: Rational) =>
-    ???
+    -rational == Rational(-rational.numer, rational.denom)
   }
 
   property("addition") = forAll { (left: Rational, right: Rational) =>
-    ???
+    left + right == Rational(left.numer * right.denom + right.numer * left.denom, left.denom * right.denom)
   }
 
   property("subtraction") = forAll { (left: Rational, right: Rational) =>
-    ???
+    left - right == Rational(left.numer * right.denom - right.numer * left.denom, left.denom * right.denom)
   }
 
   property("multiplication") = forAll { (left: Rational, right: Rational) =>
-    ???
+    left * right == Rational(left.numer * right.numer, left.denom * right.denom)
   }
 
   property("division") = forAll { (left: Rational, numer: Int, denom: Int) =>
     val right = Rational(if numer == 0 then 1 else numer, abs(denom) + 1)
```

Comment:
You already have `Zero`, `PositiveInteger` and `NegativeInteger` in `arbitraries` sample. Please use it.
Avoid conditions in the tests as much as possible. Usually, if you have conditions it means that the test could be split on two cases.
Apply the changes for all tests where it necessary. Add new test cases if necessary.

Reply:
Was not fixed.